dbcon/joblist/tupleaggregatestep.cpp

```diff
@@ -5688,14 +5688,16 @@ uint64_t TupleAggregateStep::doThreadedAggregate(ByteStream& bs, RowGroupDL* dlp
       // much memory on average
       uint32_t threads = std::max(1U, fNumOfThreads / 2);
       runners.reserve(threads);
-      for (i = 0; i < threads; ++i)
+      for (uint32_t threadNum = 0; threadNum < threads; ++threadNum)
       {
-        runners.push_back(jobstepThreadPool.invoke(ThreadedAggregateFinalizer(this, i)));
+        runners.push_back(jobstepThreadPool.invoke(ThreadedAggregateFinalizer(this, threadNum)));
       }
       jobstepThreadPool.join(runners);
     }
 
-    if (dynamic_cast<RowAggregationDistinct*>(fAggregator.get()) && fAggregator->aggMapKeyLength() > 0)
+    auto* distinctAgg = dynamic_cast<RowAggregationDistinct*>(fAggregator.get());
+    //Case 1: Query contains DISTINCT keyword and GROUP BY attributes
```

Comment:
I would change the wording so that it is clear DISTINCT comes inside aggregation functions to get distinct columns values and it is not used to get a set of distinct rows.

Reply:
yes definitely, thank you for the catch. Tbh I wasn't aware this was the case until I tested it just now, but (which makes sense) if we have a stand-alone DISTINCT this will be handled by `tupleannexstep` and we only need dedicated DistinctAggregation for nested DISTINCTs. Will annotate probably.